sites/partners/src/components/listings/PaperListingForm/sections/ListingPhotos.tsx

```diff
@@ -0,0 +1,309 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react"
+import { useFieldArray, useFormContext } from "react-hook-form"
+import {
+  t,
+  AppearanceStyleType,
+  Dropzone,
+  GridSection,
+  GridCell,
+  MinimalTable,
+  TableThumbnail,
+  StandardTableData,
+  StandardTableCell,
+  Button,
+  Drawer,
+} from "@bloom-housing/ui-components"
+import { getUrlForListingImage, CLOUDINARY_BUILDING_LABEL } from "@bloom-housing/shared-helpers"
+
+import { cloudinaryFileUploader, fieldHasError } from "../../../../lib/helpers"
+import { ListingImage, Asset } from "@bloom-housing/backend-core"
+
+const ListingPhotos = () => {
+  const formMethods = useFormContext()
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, watch, errors, clearErrors } = formMethods
+
+  const { fields, append, remove } = useFieldArray({
+    name: "images",
+  })
+  const listingFormPhotos: ListingImage[] = watch("images").sort(
+    (imageA, imageB) => imageA.ordinal - imageB.ordinal
+  )
+
+  const saveImageFields = (images: ListingImage[]) => {
+    remove(fields.map((item, index) => index))
+    images.forEach((item, index) => {
+      append({
+        ordinal: index,
+        image: item.image,
+      })
+    })
+  }
+
+  /*
+   Set state for the drawer, upload progress, images in the drawer, and more
+   */
+  const [drawerState, setDrawerState] = useState(false)
+  const [progressValue, setProgressValue] = useState(0)
+  const [latestUpload, setLatestUpload] = useState({
+    id: "",
+    url: "",
+  })
+  const [drawerImages, setDrawerImages] = useState<ListingImage[]>([])
+
+  const resetDrawerState = () => {
+    setDrawerState(false)
+    setDrawerImages([])
+  }
+
+  const savePhoto = useCallback(() => {
+    setDrawerImages([
+      ...drawerImages,
+      {
+        ordinal: drawerImages.length,
+        image: { fileId: latestUpload.id, label: CLOUDINARY_BUILDING_LABEL },
+      },
+    ])
+    setLatestUpload({ id: "", url: "" })
+    setProgressValue(0)
+  }, [drawerImages, latestUpload])
+
+  useEffect(() => {
+    if (latestUpload.id != "") {
+      savePhoto()
+    }
+  }, [latestUpload, savePhoto])
+
+  /*
+   Show list of images in the main listing form
+   */
+  const photoTableHeaders = {
+    preview: "t.preview",
+    fileName: "t.fileName",
+    primary: "t.primary",
+    actions: "",
+  }
+
+  const listingPhotoTableRows: StandardTableData = []
+  listingFormPhotos.forEach((image, index) => {
+    listingPhotoTableRows.push({
+      preview: {
+        content: (
+          <TableThumbnail>
+            <img src={getUrlForListingImage(image.image as Asset)} alt="" />
+          </TableThumbnail>
+        ),
+      },
+      fileName: { content: image.image.fileId.split("/").slice(-1).join() },
+      primary: {
+        content: index == 0 ? t("listings.sections.photo.primaryPhoto") : "",
+      },
+      actions: {
+        content: (
+          <Button
```

Comment:
This delete button is getting hidden if the uploaded image name is long. 
![image](https://user-images.githubusercontent.com/42942267/236511372-9bde1547-335a-4364-88ec-6dcb7b9d4fe2.png)
The delete button is still there, you just need to scroll. However there is no indication that that is an option. We might need to get design's opinion on what to do about that

Reply:
^ Is this blocking merge?

Reply:
Oops, thought I tagged @eajensenwa to get their thoughts. I haven't checked, but I'm guessing the same issue is in Detroit as well so it could be a follow up story if that is the case

Reply:
@ludtkemorgan I just checked in Detroit and the behavior is the same. Seems like a pattern we need to rethink if this isn't intentional. I'd recommend wrapping text before pushing action buttons to scroll, but this might have been determined by additional context I'm not aware of.

Reply:
In terms of this PR, are we okay to create a followup ticket for that since it is existing?

Reply:
Ticket for reference #3435